TorchSharp/Raw/Vector.tt

```diff
@@ -0,0 +1,153 @@
+﻿<#@ Template Language="C#" #>
+<#@ Assembly Name="System.Core" #>
+<#@ Output Extension=".generated.cs" #>
+<#@ Include File="Types.ttinclude" #>
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace TorchSharp.Raw {
+
+    internal static class Vector {
+
+<#
+foreach (var type in TorchTypeDef.Types) {
+#>
+        // Fill n elements of vector x with the value c.

```

Comment:
What do you mean with fill n elements? Are these the first n elements, or skip the first n elements? (I see that n is ptrdiff)

Reply:
`ptrdiff_t` is used like `size_t` here and in other functions, and `x` can be an offset into the vector. That is, the function will assign `c` to `n` values starting from pointer `x`. I will update the comment to clarify.

Reply:
ah! I see. For some reason I was assuming that `ptrdiff_t` was the index in `x`.